ts/src/pro/kraken.ts

```diff
@@ -1505,12 +1505,26 @@ export default class kraken extends krakenRest {
                     exception = new broad[broadKey] (errorMessage);
                 }
                 client.reject (exception, requestId);
+                this.clientRejectByRequestId (client, requestId, exception);
                 return false;
             }
         }
         return true;
     }
 
+    clientRejectByRequestId (client: Client, requestId: any = undefined, exception: any = undefined) {
+        if (!this.valueIsDefined (requestId)) {
+            return;
+        }
+        const keys = Object.keys (client.subscriptions);
```

Comment:
client.reject rejects pending futures, but you're reading from `client.subscriptions` which is not exactly the same.

Besides that, sometimes `client.subscriptions` holds boolean values, so doing `client.subscriptions[key]['reqId']` might be dangerous

Reply:
will check if there's an easier way